### PR TITLE
Small fixes for the periodogram page

### PR DESCRIPTION
**Pull request by profjsb**
- Handle the case where there is not much data observed over a very long time period, producing a more robust delta_t
- Handle the case when there is only a few data points (fixes what @dmitryduev found with a single photometry point). In that case, do not render the periodogram.
- ensure that we're only doing as many as 300,000 calculations (~<5 seconds in a modern browser)
- render the new light curve after resetting parameters.

### CODE REVIEW

**Review by dmitryduev — APPROVED** [review]
Looks good, thanks @profjsb!

File: static/js/components/Periodogram.jsx
Comment (profjsb) [inline static/js/components/Periodogram.jsx:171]: Until we have a client-side graphics card implementation, limit the total number of frequencies to 300,000. This takes at most ~5 seconds on a modern system/browser.

**Comment by dmitryduev** [comment]
Waiting on the CI, will merge once green.

**Comment by dmitryduev** [comment]
The only failure on the CI is in `test_periodogram` -- could you please look into it @profjsb?

File: skyportal/tests/frontend/test_periodogram.py
Comment (dmitryduev) [inline skyportal/tests/frontend/test_periodogram.py:74]: nice!

**Review by dmitryduev — APPROVED** [review]

**Pull request merged**
